rcp/org.testeditor.rcp4.tcltestrun/src/org/testeditor/rcp4/tcltestrun/MavenExecutor.java

```diff
@@ -49,7 +51,16 @@
 	public int execute(String parameters, String pathToPom) {
 		System.setProperty("maven.multiModuleProjectDirectory", pathToPom);
 		MavenCli cli = new MavenCli();
-		int result = cli.doMain(parameters.split(" "), pathToPom, System.out, System.err);
+		List<String> params = new ArrayList<String>();
+		params.addAll(Arrays.asList(parameters.split(" ")));
+		String mavenSettings = System.getProperty("TE.MAVENSETTINGSPATH");
+		if (mavenSettings != null) {
+			if (mavenSettings.length() > 0) {
```

Comment:
We can add a static import to `org.eclipse.xtext.xbase.lib.StringExtensions.isNullOrEmpty(String)` here to make the code more readable.